shared/js/background/declarative-net-request.js

```diff
@@ -465,6 +473,43 @@ export async function ensureServiceWorkerInitiatedRequestException () {
     })
 }
 
+/**
+ * Creates a DNR rule for ATB parameters
+ * @param {string} atb
+ * @param {object} regexFilter
+ */
+export async function setOrUpdateATBdnrRule (atb, regexFilter) {
+    if (!(atb && regexFilter)) {
+        return
+    }
+
+    // format regex for json and dnr
+    regexFilter = regexFilter.toString().replace(/\\/g, '\\').replace(/^\//, '').replace(/\/$/, '')
+
+    const atbRule = generateDNRRule({
+        id: ATB_PARAM_RULE_ID,
+        priority: ATB_PARAM_PRIORITY,
+        actionType: 'redirect',
+        redirect: {
+            transform: {
+                queryTransform: {
+                    addOrReplaceParams: [{ key: 'atb', value: atb }]
+                }
+            }
+        },
+        resourceTypes: ['main_frame'],
+        requestDomains: ['duckduckgo.com'],
+        regexFilter
+    })
+
+    if (atbRule?.id === ATB_PARAM_RULE_ID) {
```

Comment:
Why do we have this check? Won't `atbRule.id` always equal `ATB_PARAM_RULE_ID` at this point, or am I missing something?

Reply:
It's coming from ddg2dnr. It's our library, but still and external library.

Reply:
I'm probably missing something, but it seems like we create the `atbRule` Object directly above and we're sure the "id" property will equal `ATB_PARAM_RULE_ID`, so I don't understand the need for the check . Do you mean it's in case `generateDNRRule` isn't behaving like we expect?

Reply:
It works fine now. I suppose this case is covered in ddg2dnr tests anyway?

Reply:
I removed it.